src/ExtensionConfigHandler.php

```diff
@@ -352,6 +352,156 @@ protected function getSourceStorageWrapper(array $source_dirs, $subdir = Install
+   *   The config key to find the dependencies of.
+   * @param null $recursion_limit
+   *   Optionally limit the levels of recursion.
+   * @return mixed
+   */
+  public function getIndividualConfigDependancies($name, $recursion_limit = NULL) {
+    static $recursive_iterations = 0;
+    static $checked = array();
+
+    if (!isset($checked[$name])) {
+      $config_dependencies = \Drupal::config($name)->get('dependencies');
+      if ($config_dependencies && is_array($config_dependencies)) {
+        foreach ($config_dependencies as $dependency_type => $type_dependencies) {
+          // Use associative array to avoid duplicates.
+          $config_dependencies[$dependency_type] = array_combine($type_dependencies, $type_dependencies);
+        }
+
+        // Recurse to find sub-dependencies.
+        if (isset($config_dependencies['config'])) {
+          $recursive_iterations++;
+          if ($recursion_limit && $recursive_iterations < $recursion_limit) {
+            foreach ($config_dependencies['config'] as $dependency) {
+              $sub_dependencies = $this->getIndividualConfigDependancies($dependency, $recursion_limit);
+
+              // Add this dependency's dependencies to the list to be returned.
+              foreach ($sub_dependencies as $dependency_type => $type_dependencies) {
+                if (!isset($config_dependencies[$dependency_type])) {
+                  $config_dependencies[$dependency_type] = array();
+                }
+                $config_dependencies[$dependency_type] += $type_dependencies;
+              }
+            }
+          }
+          $recursive_iterations--;
+        }
+      }
+      else {
+        $config_dependencies = array();
+      }
+
+      // Config provider is an implied module dependency.
+      $config_provider = substr($name, 0, strpos($name, '.'));
+      $config_dependencies['module'][$config_provider] = $config_provider;
+
+      $checked[$name] = $config_dependencies;
+    }
+
+    return $checked[$name];
+  }
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDependentConfigSuggestions($extension, $recursion_limit = NULL) {
+    $dependants = array();
+    $listed_config = $this->getExtensionInfo($extension, 'managed');
+    $dependency_manager = \Drupal::service('config.manager')
+      ->getConfigDependencyManager();
+    foreach ($listed_config as $key) {
+      // Recursively fetch configuration entities that are dependent on this
+      // configuration entity (i.e. reverse dependencies).
+      $dependants += $this->getIndividualConfigDependants($key, $dependency_manager, $recursion_limit);
+    }
+    return $this->sortAndFilterOutput(array('config' => $dependants));
+  }
+
+  /**
+   * @param $key
+   *   The config key to find the dependancies of.
+   * @param $dependency_manager
+   *
+   * @param null $recursion_limit
+   * @return mixed
+   */
+  public function getIndividualConfigDependants($key, $dependency_manager, $recursion_limit = NULL) {
```

Comment:
As mentioned for the equivalent functions for dependencies, could we rename this and the getDependentConfigSuggestions functions to be clearer that one deals with an extension as a whole, and only the other is for a config item? There was a comment around renaming getDependentConfigSuggestions() too, so let's figure that one out first, but that one should really have `ExtensionConfig` in the name I think, whereas its this one that could just be `somethingConfigSomething`.
...and the function should be protected (+1 from me) or in the interface (I could be persuaded of this).